drivers/sensor/adt7420/adt7420.c

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright (c) 2018 Analog Devices Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <device.h>
+#include <i2c.h>
+#include <misc/byteorder.h>
+#include <misc/util.h>
+#include <kernel.h>
+#include <sensor.h>
+#include <misc/__assert.h>
+#include "adt7420.h"
+
+static int adt7420_temp_reg_read(struct adt7420_data *drv_data,
+			   u8_t reg, s16_t *val)
+{
+	if (i2c_burst_read(drv_data->i2c, drv_data->i2c_addr,
+			   reg, (u8_t *) val, 2) < 0) {
+		return -EIO;
+	}
+
+	*val = sys_be16_to_cpu(*val);
+
+	return 0;
+}
+
+static int adt7420_temp_reg_write(struct adt7420_data *drv_data,
+			    u8_t reg, s16_t val)
+{
+	u8_t buf[3] = {reg, val >> 8, val & 0xFF};
+
+	return i2c_write(drv_data->i2c, buf, sizeof(buf), drv_data->i2c_addr);
+}
+
+static int adt7420_attr_set(struct device *dev,
+			   enum sensor_channel chan,
+			   enum sensor_attribute attr,
+			   const struct sensor_value *val)
+{
+	struct adt7420_data *drv_data = dev->driver_data;
+	u8_t val8, reg = 0;
+	u16_t rate;
+	s64_t value;
+
+	if (chan != SENSOR_CHAN_AMBIENT_TEMP) {
+		return -ENOTSUP;
+	}
+
+	switch (attr) {
+	case SENSOR_ATTR_SAMPLING_FREQUENCY:
+		rate = val->val1 * 1000 + val->val2 / 1000; /* rate in mHz */
+
+		switch (rate) {
+		case 240:
+			val8 = ADT7420_OP_MODE_CONT_CONV;
+			break;
+		case 1000:
+			val8 = ADT7420_OP_MODE_1_SPS;
+			break;
+		default:
+			return -ENOTSUP;
+		}
+
+		if (i2c_reg_update_byte(drv_data->i2c, drv_data->i2c_addr,
+					ADT7420_REG_CONFIG,
+					ADT7420_CONFIG_OP_MODE(~0),
+					ADT7420_CONFIG_OP_MODE(val8)) < 0) {
+			SYS_LOG_DBG("Failed to set attribute!");
+			return -EIO;
+		}
+
+		return 0;
+	case SENSOR_ATTR_UPPER_THRESH:
+		reg = ADT7420_REG_T_HIGH_MSB;
+		/* Fallthrough */
+	case SENSOR_ATTR_LOWER_THRESH:
+		if (!reg)
+			reg = ADT7420_REG_T_LOW_MSB;
```

Comment:
Please wrap in braces

Reply:
Please bear with me, that's my first Zephyr patch. I thought that we follow Linux kernel coding style.
Braces around single line code blocks seems to be an exception...

Reply:
Yes, this is an exception to the Linux coding style.